src/Sinch/SinchClient.cs

```diff
@@ -353,5 +291,63 @@ public ISinchAuth Auth
                 new Uri(_apiUrlOverrides?.VoiceUrl ?? string.Format(VoiceApiUrlTemplate, callingRegion.Value)),
                 _loggerFactory, http);
         }
+
+        private void ValidateCommonCredentials()
+        {
+            var exceptions = new List<Exception>();
+            if (string.IsNullOrEmpty(_keyId))
+                exceptions.Add(new InvalidOperationException("keyId should have a value"));
+
+            if (string.IsNullOrEmpty(_projectId))
+                exceptions.Add(new InvalidOperationException("projectId should have a value"));
+
+            if (string.IsNullOrEmpty(_keySecret))
+                exceptions.Add(new InvalidOperationException("keySecret should have a value"));
+
+            if (exceptions.Any()) throw new AggregateException("Credentials are missing", exceptions);
+        }
+
+        private SmsClient InitSms(SinchOptions optionsObj, IHttp httpSnakeCase)
+        {
+            if (optionsObj.ServicePlanIdOptions != null)
+            {
+                _logger?.LogInformation("Initializing SMS client with {service_plan_id} in {region}",
+                    optionsObj.ServicePlanIdOptions.ServicePlanId,
+                    optionsObj.ServicePlanIdOptions.HostingRegion.Value);
+                var bearerSnakeHttp = new Http(new BearerAuth(optionsObj.ServicePlanIdOptions.ApiToken), _httpClient,
+                    _loggerFactory?.Create<Http>(),
+                    SnakeCaseNamingPolicy.Instance);
+                return new SmsClient(optionsObj.ServicePlanIdOptions.ServicePlanId,
+                    GetSmsBaseAddress(optionsObj.ServicePlanIdOptions.HostingRegion,
+                        _apiUrlOverrides?.SmsUrl),
+                    _loggerFactory, bearerSnakeHttp);
+            }
+
+            _logger?.LogInformation("Initializing SMS client with {project_id} in {region}", _projectId,
+                optionsObj.SmsHostingRegion.Value);
+            return new SmsClient(_projectId, GetSmsBaseAddress(optionsObj.SmsHostingRegion, _apiUrlOverrides?.SmsUrl),
+                _loggerFactory,
+                httpSnakeCase);
+        }
+
+        private static Uri GetSmsBaseAddress(SmsServicePlanIdHostingRegion smsServicePlanIdHostingRegion,
```

Comment:
Related to previous comment about different types for same region usage: method signature based onto region type is used to trigger `servicePlanId` vs `projectId` URL.
This should be related to `servicePlanId` usage not onto region type usage

Reply:
I'm renaming the method to better reflect underlying functionality